File: leetcode/C++/remove-all-adjacent-duplicates-in-string.cpp
class Solution {
public:
    string removeDuplicates(string s) {
        string sol;
        for(char &ch : s){
            if(sol.empty() || sol.back()!=ch) {
                sol.push_back(ch);
                continue;
            }
            while(!sol.empty() && sol.back()==ch) sol.pop_back();
        }
        return sol;
    }
};

File: leetcode/Python/remove-all-adjacent-duplicates-in-string.py
class Solution:
    def removeDuplicates(self, s: str) -> str:
        sol = []
        for c in s:
            if not sol or sol[-1] != c:
                sol.append(c)
                continue
            while sol and sol[-1] == c:
                sol.pop()
        return ''.join(sol)